diff --git a/node_modules/element-ui/lib/element-ui.common.js b/node_modules/element-ui/lib/element-ui.common.js
index 6534063..2031b19 100644
--- a/node_modules/element-ui/lib/element-ui.common.js
+++ b/node_modules/element-ui/lib/element-ui.common.js
@@ -1156,7 +1156,9 @@ var util_ = __webpack_require__(2);
         this.internalCurrentPage = 1;
       } else if (oldPage > newVal) {
         this.internalCurrentPage = newVal === 0 ? 1 : newVal;
-        this.userChangePageSize && this.emitChange();
+        // 解决删除最后一页最后一条无法自动跳转到前一页的问题
+        // this.userChangePageSize && this.emitChange();
+        this.emitChange();
       }
       this.userChangePageSize = false;
     }
@@ -18392,9 +18394,11 @@ date_table_component.options.__file = "packages/date-picker/src/basic/date-table
 
     prevMonth: function prevMonth() {
       this.date = Object(date_util_["prevMonth"])(this.date);
+      this.changeCalendar && this.changeCalendar(this.date);
     },
     nextMonth: function nextMonth() {
       this.date = Object(date_util_["nextMonth"])(this.date);
+      this.changeCalendar && this.changeCalendar(this.date);
     },
     prevYear: function prevYear() {
       if (this.currentView === 'year') {
@@ -18402,6 +18406,7 @@ date_table_component.options.__file = "packages/date-picker/src/basic/date-table
       } else {
         this.date = Object(date_util_["prevYear"])(this.date);
       }
+      this.changeCalendar && this.changeCalendar(this.date);
     },
     nextYear: function nextYear() {
       if (this.currentView === 'year') {
@@ -18409,6 +18414,7 @@ date_table_component.options.__file = "packages/date-picker/src/basic/date-table
       } else {
         this.date = Object(date_util_["nextYear"])(this.date);
       }
+      this.changeCalendar && this.changeCalendar(this.date);
     },
     handleShortcutClick: function handleShortcutClick(shortcut) {
       if (shortcut.onClick) {
@@ -42425,7 +42431,8 @@ var descriptions_row_extends = Object.assign || function (target) { for (var i =
                 }, _ref3[item.labelClassName] = true, _ref3),
                 style: item.labelStyle
               },
-              [item.props.label]
+              // 修复label slot问题
+              [item.label]
             ), h(
               'span',
               {
diff --git a/node_modules/element-ui/packages/date-picker/src/panel/date.vue b/node_modules/element-ui/packages/date-picker/src/panel/date.vue
index dc58c28..b1ae94e 100644
--- a/node_modules/element-ui/packages/date-picker/src/panel/date.vue
+++ b/node_modules/element-ui/packages/date-picker/src/panel/date.vue
@@ -1,5 +1,5 @@
 <template>
-  <transition name="el-zoom-in-top" @after-enter="handleEnter" @after-leave="handleLeave">
+  <transition @after-enter="handleEnter" @after-leave="handleLeave">
     <div
       v-show="visible"
       class="el-picker-panel el-date-picker el-popper"
