diff --git a/node_modules/element-ui/lib/element-ui.common.js b/node_modules/element-ui/lib/element-ui.common.js
index 6534063..974ab36 100644
--- a/node_modules/element-ui/lib/element-ui.common.js
+++ b/node_modules/element-ui/lib/element-ui.common.js
@@ -1156,7 +1156,9 @@ var util_ = __webpack_require__(2);
         this.internalCurrentPage = 1;
       } else if (oldPage > newVal) {
         this.internalCurrentPage = newVal === 0 ? 1 : newVal;
-        this.userChangePageSize && this.emitChange();
+        // 解决删除最后一页最后一条无法自动跳转到前一页的问题
+        // this.userChangePageSize && this.emitChange();
+        this.emitChange();
       }
       this.userChangePageSize = false;
     }
@@ -18392,9 +18394,11 @@ date_table_component.options.__file = "packages/date-picker/src/basic/date-table
 
     prevMonth: function prevMonth() {
       this.date = Object(date_util_["prevMonth"])(this.date);
+      this.changeCalendar && this.changeCalendar(this.date);
     },
     nextMonth: function nextMonth() {
       this.date = Object(date_util_["nextMonth"])(this.date);
+      this.changeCalendar && this.changeCalendar(this.date);
     },
     prevYear: function prevYear() {
       if (this.currentView === 'year') {
@@ -18402,6 +18406,7 @@ date_table_component.options.__file = "packages/date-picker/src/basic/date-table
       } else {
         this.date = Object(date_util_["prevYear"])(this.date);
       }
+      this.changeCalendar && this.changeCalendar(this.date);
     },
     nextYear: function nextYear() {
       if (this.currentView === 'year') {
@@ -18409,6 +18414,7 @@ date_table_component.options.__file = "packages/date-picker/src/basic/date-table
       } else {
         this.date = Object(date_util_["nextYear"])(this.date);
       }
+      this.changeCalendar && this.changeCalendar(this.date);
     },
     handleShortcutClick: function handleShortcutClick(shortcut) {
       if (shortcut.onClick) {
@@ -23089,6 +23095,7 @@ var form_itemvue_type_template_id_b6f3db6c_render = function() {
   return _c(
     "div",
     {
+      attrs: process.env.NODE_ENV !== 'production' && { name: _vm.prop },
       staticClass: "el-form-item",
       class: [
         {
@@ -42425,7 +42432,8 @@ var descriptions_row_extends = Object.assign || function (target) { for (var i =
                 }, _ref3[item.labelClassName] = true, _ref3),
                 style: item.labelStyle
               },
-              [item.props.label]
+              // 修复label slot问题
+              [item.label]
             ), h(
               'span',
               {
diff --git a/node_modules/element-ui/packages/date-picker/src/panel/date.vue b/node_modules/element-ui/packages/date-picker/src/panel/date.vue
index dc58c28..558f79d 100644
--- a/node_modules/element-ui/packages/date-picker/src/panel/date.vue
+++ b/node_modules/element-ui/packages/date-picker/src/panel/date.vue
@@ -1,5 +1,5 @@
 <template>
-  <transition name="el-zoom-in-top" @after-enter="handleEnter" @after-leave="handleLeave">
+  <transition>
     <div
       v-show="visible"
       class="el-picker-panel el-date-picker el-popper"
@@ -277,10 +277,12 @@
 
       prevMonth() {
         this.date = prevMonth(this.date);
+        this.changeCalendar && this.changeCalendar(this.date);
       },
 
       nextMonth() {
         this.date = nextMonth(this.date);
+        this.changeCalendar && this.changeCalendar(this.date);
       },
 
       prevYear() {
@@ -289,6 +291,7 @@
         } else {
           this.date = prevYear(this.date);
         }
+        this.changeCalendar && this.changeCalendar(this.date);
       },
 
       nextYear() {
@@ -297,6 +300,7 @@
         } else {
           this.date = nextYear(this.date);
         }
+        this.changeCalendar && this.changeCalendar(this.date);
       },
 
       handleShortcutClick(shortcut) {
@@ -399,61 +403,6 @@
         }
       },
 
-      handleEnter() {
-        document.body.addEventListener('keydown', this.handleKeydown);
-      },
-
-      handleLeave() {
-        this.$emit('dodestroy');
-        document.body.removeEventListener('keydown', this.handleKeydown);
-      },
-
-      handleKeydown(event) {
-        const keyCode = event.keyCode;
-        const list = [38, 40, 37, 39];
-        if (this.visible && !this.timePickerVisible) {
-          if (list.indexOf(keyCode) !== -1) {
-            this.handleKeyControl(keyCode);
-            event.stopPropagation();
-            event.preventDefault();
-          }
-          if (keyCode === 13 && this.userInputDate === null && this.userInputTime === null) { // Enter
-            this.emit(this.date, false);
-          }
-        }
-      },
-
-      handleKeyControl(keyCode) {
-        const mapping = {
-          'year': {
-            38: -4, 40: 4, 37: -1, 39: 1, offset: (date, step) => date.setFullYear(date.getFullYear() + step)
-          },
-          'month': {
-            38: -4, 40: 4, 37: -1, 39: 1, offset: (date, step) => date.setMonth(date.getMonth() + step)
-          },
-          'week': {
-            38: -1, 40: 1, 37: -1, 39: 1, offset: (date, step) => date.setDate(date.getDate() + step * 7)
-          },
-          'day': {
-            38: -7, 40: 7, 37: -1, 39: 1, offset: (date, step) => date.setDate(date.getDate() + step)
-          }
-        };
-        const mode = this.selectionMode;
-        const year = 3.1536e10;
-        const now = this.date.getTime();
-        const newDate = new Date(this.date.getTime());
-        while (Math.abs(now - newDate.getTime()) <= year) {
-          const map = mapping[mode];
-          map.offset(newDate, map[keyCode]);
-          if (typeof this.disabledDate === 'function' && this.disabledDate(newDate)) {
-            continue;
-          }
-          this.date = newDate;
-          this.$emit('pick', newDate, true);
-          break;
-        }
-      },
-
       handleVisibleTimeChange(value) {
         const time = parseDate(value, this.timeFormat);
         if (time && this.checkDateWithinRange(time)) {
